src/main/scala/io/iohk/ethereum/consensus/pow/PoWConsensus.scala

```diff
@@ -65,15 +83,30 @@ class PoWConsensus private (
    *
    * TODO further refactors should focus on extracting two types - one with a miner, one without - based on the config
    */
-  private[this] def startMiningProcess(node: Node): Unit = {
-    if (minerRef.isEmpty) {
+  private[this] def startMiningProcess(node: Node, blocKCreator: PoWBlockCreator): Unit = {
+    if (minerCoordinatorRef.isEmpty && mockedMinerRef.isEmpty) {
```

Comment:
shouldn't this be hidden behind the mutex?

Reply:
as in: it's redundant

Reply:
I don't understand what you mean

Reply:
the `if` statement before the mutex is subject to context switch, and since it's two conditions it will return a false positive more often than a singular check - so IMO the mutex should be locked first and the check should be performed inside the mutex only as the external one provides little security.

Reply:
So you mean, from this
```
if (minerCoordinatorRef.isEmpty && mockedMinerRef.isEmpty) {
      mutex.synchronized {
            if (minerCoordinatorRef.isEmpty && mockedMinerRef.isEmpty) {
``` 
to
```
 mutex.synchronized {
     if (minerCoordinatorRef.isEmpty && mockedMinerRef.isEmpty) {
```
?

Reply:
yep

Reply:
👍